src/gradualizer_lib.erl

```diff
@@ -170,3 +177,32 @@ get_ast_children({clauses, Clauses}) ->
 get_ast_children(Node) ->
     [_Tag, _Anno | Children] = tuple_to_list(Node),
     Children.
+
+
+%% ------------------------------------------------
+%% Type environment
+%% ------------------------------------------------
+
+-spec empty_tenv() -> tenv().
+empty_tenv() ->
+    #{module => undefined,
```

Comment:
The empty tenv is sort of a hack. It is (or should) probably only be used when the the module is not needed, but `undefined` is an atom that could actually be a module name... Not sure we can do anything about it though. Maybe omit module from the map, is that better?

Reply:
Hmm, this is a direct translation from using a record, but indeed, it seems the original use of a record was buggy in this place - https://github.com/josefs/Gradualizer/blob/212db7019c91d66e38be3ae5bc8cf35f52a1b951/src/typechecker.erl#L419. In other words, if we remove this field from the empty tenv, tests don't pass with:

```
test:27: gen_should_pass (intersection_with_any.erl)...*failed*
in function erlang:map_get/2
  called as map_get(module,#{records => #{},types => #{}})
in call from typechecker:glb/4 (src/typechecker.erl, line 414)
in call from typechecker:add_any_types_pat/2 (src/typechecker.erl, line 4058)
in call from typechecker:'-add_type_pat/4-lc$^1/1-0-'/2 (src/typechecker.erl, line 3769)
in call from typechecker:add_type_pat/4 (src/typechecker.erl, line 3769)
in call from typechecker:add_types_pats/7 (src/typechecker.erl, line 3677)
in call from typechecker:check_clause/5 (src/typechecker.erl, line 3235)
in call from typechecker:'-check_clauses/5-fun-0-'/6 (src/typechecker.erl, line 3197)
**error:{badkey,module}
  output:<<"">>
```

Which means that when the `#tenv{}` record was in use, `undefined` was the module name passed to `gradualizer_cache:get_glb()`.

Reply:
Right, let's leave it as is then. Perhaps leave a TODO comment if you want..